src/oop/programs_with_multiple_classes/medium/q10/MainTest.java

```diff
@@ -0,0 +1,133 @@
+        assertTrue(cupboard.hasModifier(varVegtables, "private"),
+                "The " + varVegtables + " attribute in your " + classCupboard + " class does not have the correct visibility modifier.");
+    }
+
+    @Test
+    public void vegetableClassHasRequiredConstructor() {
+        Class<?>[] classArguments = {boolean.class, double.class, String.class};
+        assertTrue(vegetable.hasConstructor(classArguments),
+                "Your " + classVegtable + " constructor does not have the correct parameters.");
+        assertTrue(vegetable.hasModifier(classArguments, "public"),
+                "Your " + classVegtable + " constructor does not have the correct visibility modifier.");
+    }
+
+    private static Stream<Arguments> vegetableInputProvider() {
+        return Stream.of(
+                Arguments.of(true, 20.0, "Canada"),
+                Arguments.of(false, 2.0, "USA"),
+                Arguments.of(true, 3.5, "Russia"),
+                Arguments.of(false, 5.5, "China")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("vegetableInputProvider")
+    public void vegetableConstructorInitializesValuesCorrectly(boolean isSpoiled, double weight, String countryOfOrigin) throws Throwable {
+        Object[][] arguments = {
+                {isSpoiled, boolean.class},
+                {weight, double.class},
+                {countryOfOrigin, String.class}
+        };
+        Object vegetableInstance = vegetable.createInstance(arguments);
+        _assertEquals(isSpoiled, vegetable.getFieldValue(vegetableInstance, varSpoiled),
+                "Your " + classVegtable + " constructor does not correctly initialize the " + varSpoiled + " attribute.");
+        _assertEquals(weight, vegetable.getFieldValue(vegetableInstance, varWeight),
+                "Your " + classVegtable + " constructor does not correctly initialize the " + varWeight + " attribute.");
+        _assertEquals(countryOfOrigin, vegetable.getFieldValue(vegetableInstance, varOrigin),
+                "Your " + classVegtable + " constructor does not correctly initialize the " + varOrigin + " attribute.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("vegetableInputProvider")
+    public void correctVegetableEatingMethod(boolean isSpoiled, double weight, String countryOfOrigin) throws Throwable {
+        Object[][] arguments = {
+                {isSpoiled, boolean.class},
+                {weight, double.class},
+                {countryOfOrigin, String.class}
+        };
+        Object vegetableInstance = vegetable.createInstance(arguments);
+        Object output = vegetable.callMethod(methodName, vegetableInstance);
+        _assertEquals(weight - 0.5, vegetable.getFieldValue(vegetableInstance, varWeight),
+                "Your " + classVegtable + " " + methodName + " method does not decrement the " + varWeight + " by 0.5.");
+        assertNull(output, "Your " + methodName + " method should not return anything.");
```

Comment:
Check the method return type instead of asserting that the method output is null, please.